src/buffer.rs

```diff
@@ -206,6 +225,12 @@ impl<'a, P: Pixel + 'a> Iterator for EnumeratePixelsMut<'a, P> where P::Subpixel
     }
 }
 
+impl <'a, P: Pixel + 'a> ExactSizeIterator for EnumeratePixelsMut<'a, P> where P::Subpixel: 'a {
```

Comment:
Can the entire where clause and all bounds on `P` be replaced with `where Self: Iterator`?

Reply:
Looks like the right thing.